pyam/core.py

```diff
@@ -499,7 +500,13 @@ def check_aggregate(self, variable, components=None, units=None,
             components = self.filter(variable='{}|*'.format(variable),
                                      level=0).variables()
 
-        # filter and groupby data, use `pd.Series.align` for machting index
+        if not len(components):
```

Comment:
The reason I'd favour not raising an error is that you then have to wrap all your checks in try except blocks if you want to just throw a table which has variables with no components at it. If you just print a message and return None then things can keep running. Happy either way though and happy to put the try except in the `internal_consistency` method so the logic is kept separate.

Reply:
maybe @danielhuppmann has an opinion. I saw another function he wrote with a option like `fail_on_error` or something..

Reply:
I'd rather see everything that is internally inconsistent at once rather than get an error message every time the internal-consistency finds an issue, and I have to fix it and run the function again

Reply:
ok so I'll leave it then